HIP/hip-655.md

```diff
@@ -0,0 +1,106 @@
+- During token transfer validation, the network finds that the receivers have not associated this token with their accounts. Instead of rejecting this call with a failure code, the new behavior will store these tokens in the Airdrop-Sender's account in a new data structure called the `unsolicited-tokens-outbox`. Similarly, there is a concept of an `unsolicited-tokens-inbox` (or the incoming spam inbox) in the Airdrop-Recipients' accounts. And an entry is added in each of these recipient accounts registering the proposed transfers.
+- Network resources are used for storing these entries in both of these accounts (sender and receiver). The `CryptoTransfer` call will have an additional fee to cover the rent for these resources for one auto-renewal cycle. And the Airdrop-Sender will have to pay for the renewal of these resources in their normal auto-renewal process while these tokens remain in the unsolicited-outbox.
+- This unsolicited-inbox information is available on the mirror nodes for the Airdrop-Recipients. And Wallets and Explorers can optionally show the content of these in their user interfaces.
+- If the Airdrop-Recipient wants to accept the airdrop, they will invoke a new HAPI API - say, `AcceptUnsolicitedToken` - to indicate their acceptance of this airdrop. Wallets and third-party dApps can facilitate this process through intuitive user journeys for the users.
+- Upon receiving this `AcceptUnsolicitedToken` HAPI command, the network will actually move the token from Airdrop-Sender's account to the Airdrop-Receiver's account and also add a token-association entry for this token on the receiver's account automatically. From this point onwards, the token behaves exactly as if the Airdrop-Recipient had manually created a token association and then received the token to the account.
+- The Airdrop-Recipient is responsible for renewal of the token association as part of their normal renewal after this point. Please note that the tokens are in the Airdrop-Sender's control until the Airdrop-Recipient calls the `AcceptUnsolicitedToken` API. As a result, they can also delete or burn these tokens while they are in the outbox. Doing so will remove the entries from the Unsolicited-inbox and Unsolicited-outbox automatically. Please note that these are conceptual names and the actual APIs/protobufs and REST APIs (mirror node) will be designed as this HIP develops further.
+
+### New APIs/UIs:
+
+- Services:
+    - `AcceptUnsolicitedToken` API
+- Mirror node:
+    - GET `unsolicitedInbox` query for the Airdrop-Recipient to get content of their unsolicited-inbox.
+    - GET `unsolicitedOutbox` query for the Airdrop-Sender to get content of their unsolicited-outbox.
+- Wallets and Explorers
+    - Wallets and Explorers should provide a way for users to be able to see the unsolicited-inbox and accept the tokens that they wish to accept
+
+### Other considerations
```

Comment:
Good questions.
I imagine if a token is in `unsolicited-token-inbox` then
1. `IERC20(airdroppedToken).balanceOf(AirdropRecipient)` => `0`
2. IERC20(airdroppedToken).balanceOf(AirdropSender) 
  - (either) => balance of sender including the token. Conceptually you could view it as until a recipient has accepted an airdrop the sender does does still own the tokens.
  - (or) => balance of sender including the token. Conceptually this would be like compartmentalizing it. Whiles in an outbox it doesn't count towards the main balance box. 

🤔

Reply:
I would say while the tokens airdropped are on "PENDING" status they don't count for either balances of:
IERC20(airdroppedToken).balanceOf(address). AirdropRecipient or AirdropSender.
why: 
for `AirdropRecipient` they are not associated with the token, if we assume they have a positive balance, some smart-contracts could use airdropped tokens to validate a condition, (feature or bug), and if after the "PENDING" is moved by the sender to "DELETED" feels  more like a "bug" to me and inconsistent.

Also, the balance should not be on the "sender" balance either since they have already being "airdropped" but siting on kind of a Escrow.

That is why I like the "Airdrops" to be implemented like "Optimism" and "Arbitrum", where they have to be claimed by the recipient, but also I understand the use cases for where actual airdrops might be needed.